Ginger/Ginger/Actions/ActionEditPages/ActExcelEditPage.xaml.cs

```diff
@@ -112,20 +116,44 @@ private void ViewDataButton_Click(object sender, RoutedEventArgs e)
         {
             ContextProcessInputValueForDriver();
 
-            DataTable dt = mAct.GetExcelSheetData(null);
-            if (dt != null)
-                ExcelDataGrid.ItemsSource = dt.AsDataView();
+            DataTable dt = GetExcelSheetData(true);
+            if (dt == null)
+            {
+                Reporter.ToUser(eUserMsgKey.ExcelInvalidFieldData);
+                return;
+            }
+            ExcelDataGrid.ItemsSource = dt.AsDataView();
         }
 
         private void ViewWhereButton_Click(object sender, RoutedEventArgs e)
         {
+            if (!mAct.CheckMandatoryFieldsExists(new List<string>() { 
+                nameof(mAct.CalculatedFileName), nameof(mAct.CalculatedSheetName),  nameof(mAct.CalculatedFilter)}))
```

Comment:
you tested getting the calculated fields value on edit page without running the Action first?

Reply:
yes